src/parquet/column/reader.h

```diff
@@ -201,6 +206,33 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   return total_values;
 }
 
+template <typename DType>
+inline int64_t TypedColumnReader<DType>::Skip(int64_t num_rows_to_skip) {
+  int64_t rows_to_skip = num_rows_to_skip;
+  while (HasNext() && rows_to_skip > 0) {
+    // If the number of rows to skip is more than the number of undecoded values, skip the
+    // Page.
+    if (rows_to_skip > (num_buffered_values_ - num_decoded_values_)) {
+      rows_to_skip -= num_buffered_values_ - num_decoded_values_;
+      num_decoded_values_ = num_buffered_values_;
+    } else {
+      // We need to read this Page
+      // Jump to the right offset in the Page
+      int64_t values_read = 0;
+      std::vector<uint8_t> vals(
```

Comment:
The main difficulty here is that we need to count the number of `values` in a  `DATA_PAGE` using the `definition levels`. 
I modified the patch by using memory via the allocator and reduced the footprint via chunking.
Another optimization which is probably not worth the effort is to implement a `Skip` inside the decoders. This will help avoid reading the repetition levels and values after reading the definition levels.